go-controller/pkg/ovn/pods.go

```diff
@@ -412,6 +464,77 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 		}
 	}
 
+	// add src-ip routes to GR if external gw annotation is set
+	routingExternalGWs, err := oc.getRoutingExternalGWs(pod.Namespace)
+	if err != nil {
+		return err
+	}
+	if routingExternalGWs != nil {
+		gr := "GR_" + pod.Spec.NodeName
+		for _, v := range routingExternalGWs {
+			var mask string
+			gw := v.String()
+			for _, podIPNet := range podIfAddrs {
+				podIP := podIPNet.IP.String()
+				if utilnet.IsIPv6(net.ParseIP(podIP)) {
+					mask = IPv6FullMask
+				} else {
+					mask = IPv4FullMask
+				}
+				_, stderr, err := util.RunOVNNbctl("--may-exist", "--policy=src-ip", "--ecmp",
+					"lr-route-add", gr, podIP+mask, gw)
+				if err != nil {
+					return fmt.Errorf("unable to add external gw src-ip route to GR router, stderr:%q, err:%v", stderr, err)
+				}
+				nsInfo, err := oc.waitForNamespaceLocked(pod.Namespace)
+				if err != nil {
+					return err
+				}
+				if nsInfo.podExternalRoutes[podIP] == nil {
+					nsInfo.podExternalRoutes[podIP] = make(map[string]string)
+				}
+				nsInfo.podExternalRoutes[podIP][gw] = gr
+				nsInfo.Unlock()
+			}
+		}
+	}
```

Comment:
maybe `} else if config.Gateway.DisableSNATMultipleGWs {`

Reply:
@trozet ^ , I think I had an else here at some point but you suggested to break it?

Reply:
so basically this is saying if snat is disabled for multiple gateways and this pod being added does not have multiple gateways, then add a per pod snat. IF disableSNATMultipleGws was false, we would have configured the blanket SNAT in gateway init. So i think this behavior is right (just double checking for my own sanity here). I think in that case it is fine to move it to else if as @dcbw suggests. Maybe just add some comments stating the logic here, to make things clearer.

Reply:
done